src/shared/copy.c

```diff
@@ -795,6 +796,66 @@ static int prepare_nocow(int fdf, const char *from, int fdt, unsigned *chattr_ma
         return 0;
 }
 
+/* Copies fs-verity status.  May re-open fdt to do its job. */
+static int copy_fs_verity(int fdf, int *fdt) {
+        int r;
+
+        assert(fdf >= 0);
+        assert(fdt);
+        assert(*fdt >= 0);
+
+        r = fd_verify_regular(fdf);
+        if (r < 0)
+                return r;
+
+        struct fsverity_descriptor desc = {};
+        struct fsverity_read_metadata_arg read_arg = {
+                .metadata_type = FS_VERITY_METADATA_TYPE_DESCRIPTOR,
+                .buf_ptr = (uintptr_t) &desc,
```

Comment:
This may cause build failure on some arch or when built with musl or so.
Please use `__u64` or `typeof(...)`.

Reply:
I think this is fine. We use `uintptr_t` in various places. If it turns out to be a problem with musl, musl folks will submit a patch.